src/mono/mono/mini/mini-arm64.c

```diff
@@ -3944,10 +3944,19 @@ mono_arch_output_basic_block (MonoCompile *cfg, MonoBasicBlock *bb)
 				break;
 			}
 
-			if (dreg != sreg1)
-				arm_neon_mov (code, dreg, sreg1);
-			
-			arm_neon_ins_e(code, t, dreg, sreg2, ins->inst_c0, 0);
+			if (dreg != sreg1) {
+				if (dreg != sreg2) {
+					arm_neon_mov (code, dreg, sreg1);
+					arm_neon_ins_e(code, t, dreg, sreg2, ins->inst_c0, 0);
+				} else {
+					arm_neon_mov (code, NEON_TMP_REG, sreg1);
+					arm_neon_ins_e(code, t, NEON_TMP_REG, sreg2, ins->inst_c0, 0);
+					arm_neon_mov (code, dreg, NEON_TMP_REG);
```

Comment:
You also need to increase max. length of these instructions in `cpu-arm64.mdesc`

Reply:
Oops, good catch!